### ISSUE TITLE
some SUSPICION of the result

### ISSUE THREAD

**Issue by David-Ttao** [aikunyi/FreTS#8]
I used the patchtst source code, ETTh1 data set, trained for 5 rounds, and modified the scaler in dataloader to MinMax (consistent with the setting in the article). The results are shown below. Among them, mae is lower than FreTS0.065 only when lookback is 192. The rest exceed the results provided by the authors. So I suspect that the result is not solid enough, the story is fancy, but the actual model is worth discussing.

![1710933383738](https://github.com/aikunyi/FreTS/assets/82387560/fdad9288-d462-4d2a-868d-ac9f00d74b28)
![1710933390505](https://github.com/aikunyi/FreTS/assets/82387560/f595de0a-cbf6-44fa-9159-576adfb87c1c)
![1710933511721(1)](https://github.com/aikunyi/FreTS/assets/82387560/2118b81c-71e5-420f-b858-b9a989ffd229)
![1710933520116](https://github.com/aikunyi/FreTS/assets/82387560/5bec68b8-a7bc-4d98-b0c3-122e167da3d1)

**Comment by aikunyi** [comment]
this is our results
![20240320221547](https://github.com/aikunyi/FreTS/assets/63833572/9924178b-5ffe-4e81-b4ea-8cfe0156a8c0)

**Comment by David-Ttao** [comment]
Hello There is no convincing reason why my baseline results are better than yours. And can you explain this? The second line of code does not use traindata's data features to normalize the global features. Instead, it normalizes with global features. The first line of code does not make any sense, which will cause the test set information to leak into the training set.
![1711455905244](https://github.com/aikunyi/FreTS/assets/82387560/6eb4f7ab-cc50-47ce-9ee4-183413e2a30c)
In the normal benchmark, normalization is done like this:
![1711456003464](https://github.com/aikunyi/FreTS/assets/82387560/97c38250-dd53-4920-aeb7-890bb5dd21f0)
only fit **train_data**, and transform on **df_data**. NOT fit_transform on **df_data**

**Comment by aikunyi** [comment]
Thanks for your reminder, we've revised it. If you have time, you are welcome to run FreTS with the normalization method like the normal benchmark. Thanks.

**Comment by Joeland4** [comment]
> Hello There is no convincing reason why my baseline results are better than yours. And can you explain this? The second line of code does not use traindata's data features to normalize the global features. Instead, it normalizes with global features. The first line of code does not make any sense, which will cause the test set information to leak into the training set. ![1711455905244](https://private-user-images.githubusercontent.com/82387560/316858757-6eb4f7ab-cc50-47ce-9ee4-183413e2a30c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxODkyODEsIm5iZiI6MTcxNjE4ODk4MSwicGF0aCI6Ii84MjM4NzU2MC8zMTY4NTg3NTctNmViNGY3YWItY2M1MC00N2NlLTllZTQtMTgzNDEzZTJhMzBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDA3MDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmMGVlMTAxMTQ0M2JkNWYzZTFlMDkwYWRhODIzNDE4ZDgzMThlNGRmYmVjODFjZTA5YzdjNDIxNDFjNTQ4NjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NDA0sOdr5URnbwJC8atTLcO5wsPrqeOx8OW6sVsbU80) In the normal benchmark, normalization is done like this: ![1711456003464](https://private-user-images.githubusercontent.com/82387560/316859488-97c38250-dd53-4920-aeb7-890bb5dd21f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxODkyODEsIm5iZiI6MTcxNjE4ODk4MSwicGF0aCI6Ii84MjM4NzU2MC8zMTY4NTk0ODgtOTdjMzgyNTAtZGQ1My00OTIwLWFlYjctODkwYmI1ZGQyMWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDA3MDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYWQzYjNjYmFiOWM2YzM2YzZmNDZjZGZkOTE3YTAwNGUyNzFjYmFkN2U2MjEyOTI2Y2RjNjYyZjY1OTBhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CZXCU-9nSv8rIBgF2I8UUaGvNeW34EsYb3B7iejl6UE) only fit **train_data**, and transform on **df_data**. NOT fit_transform on **df_data**

I have another two problems. first is why the results MAE and RMSE are not consistent with that in other SOTA papers such as DLinear or iTransformer？
![图片1](https://github.com/aikunyi/FreTS/assets/51989540/45d5c07e-f877-47c2-94af-260c0d74d4cb)
 The second is Did you really do the frequency MLP of Channel， the most intuitive thing is that there should be permutation invariance in channels. The biggest problem in your code is 
![image](https://github.com/aikunyi/FreTS/assets/51989540/58b38925-0a13-49b3-9d05-523f0b5eb9a3) in run_longExp.py
![image](https://github.com/aikunyi/FreTS/assets/51989540/1645e2dd-1e64-48a6-a113-7757b8f21c65) in FreTS.py
You can't do channel MLP in your model code at all, This is obviously not a mistake that can go undetected during the  experiment!   @aikunyi @David-Ttao

**Comment by ambitious121** [comment]
> > 您好：没有令人信服的理由说明我的基准结果比您的好。您能解释一下吗？第二行代码不使用 traindata 的数据特征来规范化全局特征。相反，它使用全局特征进行规范化。第一行代码没有任何意义，这将导致测试集信息泄漏到训练集中。![1711455905244](https://private-user-images.githubusercontent.com/82387560/316858757-6eb4f7ab-cc50-47ce-9ee4-183413e2a30c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxODkyODEsIm5iZiI6MTcxNjE4ODk4MSwicGF0aCI6Ii84MjM4NzU2MC8zMTY4NTg3NTctNmViNGY3YWItY2M1MC00N2NlLTllZTQtMTgzNDEzZTJhMzBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDA3MDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmMGVlMTAxMTQ0M2JkNWYzZTFlMDkwYWRhODIzNDE4ZDgzMThlNGRmYmVjODFjZTA5YzdjNDIxNDFjNTQ4NjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NDA0sOdr5URnbwJC8atTLcO5wsPrqeOx8OW6sVsbU80)在正常的基准测试中，归一化是这样的：![1711456003464](https://private-user-images.githubusercontent.com/82387560/316859488-97c38250-dd53-4920-aeb7-890bb5dd21f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxODkyODEsIm5iZiI6MTcxNjE4ODk4MSwicGF0aCI6Ii84MjM4NzU2MC8zMTY4NTk0ODgtOTdjMzgyNTAtZGQ1My00OTIwLWFlYjctODkwYmI1ZGQyMWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDA3MDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYWQzYjNjYmFiOWM2YzM2YzZmNDZjZGZkOTE3YTAwNGUyNzFjYmFkN2U2MjEyOTI2Y2RjNjYyZjY1OTBhNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CZXCU-9nSv8rIBgF2I8UUaGvNeW34EsYb3B7iejl6UE)只拟合 **train_data**，并在 **df_data** 上转换。NOT fit_transform on **df_data**
> 
> 我还有另外两个问题。首先是为什么 MAE 和 RMSE 的结果与其他 SOTA 论文（如 DLinear 或 iTransformer）中的结果不一致？![图片1](https://private-user-images.githubusercontent.com/51989540/331977268-45d5c07e-f877-47c2-94af-260c0d74d4cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwMDg2NTgsIm5iZiI6MTcyNzAwODM1OCwicGF0aCI6Ii81MTk4OTU0MC8zMzE5NzcyNjgtNDVkNWMwN2UtZjg3Ny00N2MyLTk0YWYtMjYwYzBkNzRkNGNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDEyMzIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2ZGY2YzMxZDdhMzg4ZDVkMzU4M2M1ZGNhMjc0ZmY1MjE0OGFjNjJkNjVmNjg4NzY4NmFlZTkwYTRhMmExMmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eneZ0Ah_wFoJNHEKMnKWMAiHqw1uvk8FEODYLrOez04)第二个是你真的做过 Channel 的频率 MLP 吗，最直观的是 channels 中应该有排列不变性。你的代码![image](https://private-user-images.githubusercontent.com/51989540/331978927-58b38925-0a13-49b3-9d05-523f0b5eb9a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwMDg2NTgsIm5iZiI6MTcyNzAwODM1OCwicGF0aCI6Ii81MTk4OTU0MC8zMzE5Nzg5MjctNThiMzg5MjUtMGExMy00OWIzLTlkMDUtNTIzZjBiNWViOWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDEyMzIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4MGU3MjdjNjNhNDFlMzlkNGEwMzNiYjdiMmMxN2VhZjgyZDM5NGQ4OGQyNzY4MTU0ODVlM2FkMmZlMjQ1ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fa_jX0m59xNnoFbGyr7H8TKKwOeeFuM8z3YH0aewfAs)中最大的问题是run_longExp.py ![image](https://private-user-images.githubusercontent.com/51989540/331979087-1645e2dd-1e64-48a6-a113-7757b8f21c65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwMDg2NTgsIm5iZiI6MTcyNzAwODM1OCwicGF0aCI6Ii81MTk4OTU0MC8zMzE5NzkwODctMTY0NWUyZGQtMWU2NC00OGE2LWExMTMtNzc1N2I4ZjIxYzY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDEyMzIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzNjE3ODJhNzc2NWUyYjRiODlmNzEzOGU3M2NjYzhjMzhlN2I2ZDAwZmNmMzQzMWQ0Zjc1YTQxMmE2YzRlZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MDZvvj4ellUa1kzfGiteEXkjtljZlv3ZaRYeSTX0HXQ) FreTS.py 你根本不能在你的模型代码中做通道 MLP，这显然不是一个在实验过程中可以不被发现的错误！

他这个确实没有做通道的mlp我把代码改的可以进行通道mlp之后效果反而变差了。

**Comment by zysongg** [comment]
您好，我也有些疑问。他这个指标为什么没有和Autoformer类对齐，怎么都0.00+了？

**Comment by ManishIISc** [comment]
> this is our results ![20240320221547](https://github.com/aikunyi/FreTS/assets/63833572/9924178b-5ffe-4e81-b4ea-8cfe0156a8c0)

![Image](https://github.com/user-attachments/assets/f8c05a4c-a407-49d4-aa8f-bc46d705db1e)
There is a disparity in the results, even though the same hyperparameters and exact code were used
